src/ui/hooks/recordings.ts

```diff
@@ -105,7 +107,16 @@ const GET_MY_RECORDINGS = gql`
 
 export function useGetRecordingId() {
   const { id } = useRouter().query;
-  return Array.isArray(id) ? id[0] : id!;
+  return extractIdAndSlug(id).id!;
+}
+
+export function getRecordingURL(recording: Recording) {
+  let slug = "";
+  if (recording.title) {
+    slug = `${slugify(recording.title).toLowerCase()}--`;
```

Comment:
There's also a related part that parses URL (split by `--`) down below, that is in a different place, with no shared locality in any way, even though they work with the same assumption (namely, what's the expected URL format).

I'd recommend making getting and parsing URL closer to each other (in some same utils file e.g.) or ever part of some static methods on a helper class. Updating anything with URLs will require weird digging through the codebase and hope we didn't forget to update it somewhere.

(this is a non-blocking suggestion though)

Reply:
At some point, we will have to consolidate different pages, routes and URL patterns that exist, that are currently all hardcoded in different places, so a step in that general direction is a good thing either way.

Reply:
Yeah, it might feel silly to extract `--` to a named constant like `URL_SEPARATOR` but even something as simple as that and importing it from the same place would help towards making the link discoverable 👍 , but also making generator/parser functions that live next to each other works for me!